backend/src/oe/api/op-blockspan.service.ts

```diff
@@ -0,0 +1,140 @@
+      return confirmedBlocks;
+    } catch (e) {
+      throw new Error(`${UUID} OpBlockSpanApiService.$getConfirmedBlockSpanList: error while filtering confirmed block span list: ${e instanceof Error ? e.message : e}`);
+    }
+  }
+
+  public async $generateBlockSpanDetailedList(UUID: string, endBlockHeight: number, span: number, blockSpanCount: number): Promise<BlockSpanDetails[]> {
+    try {
+      const blockSpanList = this.$getBlockSpanList(
+        UUID,
+        endBlockHeight,
+        span,
+        blockSpanCount
+      );
+      const confirmedBlockSpanList = await this.$getConfirmedBlockSpanList(UUID, blockSpanList);
+      if (!confirmedBlockSpanList.length) {
+        return [];
+      }
+
+      const blockHeadersList = await opBlockHeaderService.$getBlockHeadersByHeights(
+        UUID,
+        Array.from(new Set(confirmedBlockSpanList))
+      );
+
+      let index = 0;
+      const result: BlockSpanDetails[] = [];
+      while (index < blockHeadersList.length - 1) {
+        result.push({
+          startBlock: {
+            height: blockHeadersList[index].height,
+            hash: blockHeadersList[index].current_block_hash,
+            mediantime: blockHeadersList[index].mediantime,
+            timestamp: blockHeadersList[index].timestamp,
+          },
+          endBlock: {
+            height: blockHeadersList[++index].height,
+            hash: blockHeadersList[index].current_block_hash,
+            mediantime: blockHeadersList[index].mediantime,
+            timestamp: blockHeadersList[index].timestamp,
+          },
+        });
+      }
+      return result;
+    } catch (e) {
+      throw new Error(`${UUID} OpBlockSpanApiService.$generateBlockSpanDetailedList: error while generating detailed block span list: ${e instanceof Error ? e.message : e}`);
+    }
+  }
+
+  public async $getBlockSpanDetailedList(
+    UUID: string,
+    endBlockHeight: number,
+    span: number,
+    blockSpanCount: number,
+    withNbdrStatistics: boolean
+  ): Promise<BlockSpanDetails[] | BlockSpanDetailsWithNbdr[] | NbdrStatisticsError> {
+    try {
+      const blockSpanDetailedList = await this.$generateBlockSpanDetailedList(
+        UUID,
+        endBlockHeight,
+        span,
+        withNbdrStatistics && blockSpanCount !== -1 ? blockSpanCount + 100 : blockSpanCount
+      );
+      if (!withNbdrStatistics) {
+        return blockSpanDetailedList;
+      }
+
+      for (let index = 100; index < blockSpanDetailedList.length; index++) {
+        const blockSpanListForStat = blockSpanDetailedList.slice(index - 100, index);
+        const statistics = opStatisticsService.calculateStatisticsWithBlockSpan(blockSpanListForStat, span);
+        if ('error' in statistics) {
+          return statistics;
+        }
+        blockSpanDetailedList[index]['nbdr'] = { ...statistics.nbdr };
+      }
+
+      if (blockSpanCount === -1) {
+        return blockSpanDetailedList;
+      }
+
+      if (blockSpanDetailedList.length > 100) {
```

Comment:
this condition seems always to be true: if numberOfSpans = -1, then blockSpanDetailedList.length = `Number.MAX_VALUE`, otherwise, it will be `blockSpanCount + 100`, given that numberOfSpans is Positive, this means, that numberOfSpans is always `>=101`

Reply:
overall statistics code is hard to follow: I can't stop thinking, that this code assumes, that there are >100 confirmed blocks in the chain. But what if tomorrow  we will deploy new instance of bitcoin node and op-energy? then this node will start chain synchronization filling everything from ground up. This will mean, that there will be 1..N of confirmed blocks. What results will be given by the current code, that is dancing around value of 100? Or what is tomorrow there will be new bitcoin network introduced (similar to signet), which will start chain from the very beginning...